_posts/2022-02-02-#748.md

```diff
@@ -10,19 +10,47 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- [libsecp256k1](https://github.com/bitcoin-core/secp256k1/) is the library that Bitcoin Core uses for low-level cryptographic operations such as signing and verification.
+
+- This PR adds usage examples for several common use cases: ECDSA signatures, Schnorr signatures, and (elliptic curve) Diffie-Hellman key exchange. There are no existing usage examples in the library.
+
+- The examples are intended to clarify the API by supplementing the documentation in the [various header files](https://github.com/bitcoin-core/secp256k1/tree/master/include).
+
 ## Questions
+
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
--->
+2. Can you compile and run the example (note: you must run `./configure` with the `--enable-examples` flag)?
+3. Why do the examples demonstrate how to obtain randomness? Is this a good idea?
+4. Are those the correct ways to obtain randomness on each of the supported operating systems?
+5. Can you follow the examples?
+6. Is there anything missing in the examples (e.g. `context_randomize`, `ec_seckey_verify`, return value check, setting secrets to 0, etc...)?
+7. Are the comments in the examples helpful?
+8. Are the comments sufficient (e.g. ECDSA takes message hash, not message)?
+9. Is the build system correctly adjusted for the examples? Are examples disabled by default? Does the `configure` output show whether examples are enabled?
+
+## Further Questions
 
+These questions are intended to guide deeper exploration of libsecp256k1 and may help in answering above questions.
+
+1. What's a `secp256k1_context`? What role do they play in the API?
+2. What does it mean for a context to be "initialized for signing" or "initialized for verification"?
+3. What is the probability that `ec_seckey_verify` fails given a uniformly random input byte string?
+4. Can you give an example for a 33-byte string that results in failure when given to `ec_pubkey_parse`?
+5. How does `secp256k1_schnorrsig_sign` protect against the possibility of nonce reuse?
+6. What does it mean for a public key to be "X-only"? Why does the Schnorr example (unlike ECDSA) use "X-only" keys?
+7. `schnorrsig_sign` takes a 32-byte message. What are two ways to sign a message that is not 32 bytes.
+8. Can you give an example for a 32-byte string that results in failure when given to `xonly_pubkey_parse`?
+9. What messages are actually exchanged (across the network) during an execution of the ECDH protocol?
+10. What does "randomizing" a `secp256k1_context` do? How does that protect against side-channel attacks?
+11. What does the `auxiliary_rand` argument to `schnorrsig_sign` do?
+12. Bonus question: how could an eavesdropping attacker detect whether two parties are engaging in an ECDH protocol?
```

Comment:
Nit: I don't think this one deserves special callout as a "Bonus question" anymore

Reply:
Done